api/app/signals/apps/api/serializers/attachment.py

```diff
@@ -14,7 +14,31 @@
 from signals.apps.signals.models import Attachment, Signal
 
 
-class SignalAttachmentSerializer(HALSerializer):
+class SignalAttachmentSerializerMixin:
+    def create(self, validated_data):
+        attachment = Signal.actions.add_attachment(validated_data['file'], self.context['view'].get_signal())
+
+        if self.context['request'].user:
+            attachment.created_by = self.context['request'].user.email
+            attachment.save()
+
+        return attachment
+
+    def validate_file(self, file):
+        if file.size > settings.API_MAX_UPLOAD_SIZE:
```

Comment:
This check should be part of the `UploadScannerService` (Not a blocker for now but it would be nice to move it)

Reply:
Will be refactored in a new PR